docs/design/ovs-pipeline.md

```diff
@@ -53,62 +52,129 @@
+|               |             |                           | 0b00000100    | PacketInNPStoreDenyOperation    | Indicating the corresponding connection has been dropped or rejected.                    |
+| NXM_NX_REG1   | bits 0-31   | TargetOFPortField         |               |                                 | Egress OVS port of packet.                                                               |
+| NXM_NX_REG2   | bits 0-31   | SwapField                 |               |                                 | Swap values in flow fields in OpenFlow actions.                                          |
+|               |             | PacketInTableField        |               |                                 | OVS table where the packet is decided to be sent to controller.                          |
+| NXM_NX_REG3   | bits 0-31   | EndpointIPField           |               |                                 | Field where stores selected Service Endpoint IPv4 address.                               |
+|               | bits 0-31   | APConjIDField             |               |                                 | Field to store Conjunction ID for Antrea Policy.                                         |
+| NXM_NX_REG4   | bits 0-15   | EndpointPortField         |               |                                 | Field stores IPv4 address of a Service's selected Endpoint.                              |
+|               | bits 16-18  | ServiceEPStateField       | 0b001         | EpToSelectRegMark               | Packet needs to do Service Endpoint selection.                                           |
+|               | bits 16-18  | ServiceEPStateField       | 0b010         | EpSelectedRegMark               | Packet has done Service Endpoint selection.                                              |
+|               | bits 16-18  | ServiceEPStateField       | 0b011         | EpToLearnRegMark                | Packet has done Service Endpoint selection and the selected Endpoint needs to be cached. |
+|               | bits 0-18   | EpUnionField              |               |                                 | The union value of EndpointIPField and EndpointPortField.                                |
+|               | bit  19     |                           | 0b1           | ToNodePortAddressRegMark        | Packet is destined to a Service of type NodePort.                                        |
+|               | bit  20     |                           | 0b1           | AntreaFlexibleIPAMRegMark       | Packet is from local Antrea IPAM Pod.                                                    |
+|               | bit  20     |                           | 0b0           | NotAntreaFlexibleIPAMRegMark    | Packet is not from local Antrea IPAM Pod.                                                |
```

Comment:
I don't think you need to cover both values for "boolean" marks like this one. For example, you don't seem to be doing it for `ToNodePortAddressRegMark`?

Reply:
Yes, I only cover both values when both values are used in flows.